app/lib/bk/compat/parsers/bitrise/slack.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module BK
+  module Compat
+    module BitriseSteps
+      # Handles the translation of slack notifications
+      class Translator
+        def translate_slack(inputs)
+          warnings = generate_warnings(inputs)
+
+          if error_config?(inputs)
+            create_dual_notifications(inputs, warnings)
+          else
+            create_single_notification(inputs, warnings)
+          end
```

Comment:
you could create a single notification and, if the second one is necessary, modify the one returned by the single one and add the other one (which would mean less code duplication)